Managers/WindowMan.h

```diff
@@ -0,0 +1,333 @@
+		/// </summary>
+		/// <param name="updateInfoOfDisplayPrimaryWindowIsAt">Whether to update the stored info of the display the primary window is currently positioned at.</param>
+		void MapDisplays(bool updateInfoOfDisplayPrimaryWindowIsAt = true);
+
+		/// <summary>
+		/// Gets the horizontal resolution of the display the primary game window is currently positioned at.
+		/// </summary>
+		/// <returns>The horizontal resolution of the display the primary game window is currently positioned at.</returns>
+		int GetPriaryWindowDisplayWidth() const { return m_PrimaryWindowDisplayWidth; }
+
+		/// <summary>
+		/// Gets the vertical resolution of the display the primary game window is currently positioned at.
+		/// </summary>
+		/// <returns>The vertical resolution of the display the primary game window is currently positioned at.</returns>
+		int GetPrimaryWindowDisplayHeight() const { return m_PrimaryWindowDisplayHeight; }
+
+		/// <summary>
+		/// Gets whether the game resolution was changed.
+		/// </summary>
+		/// <returns>Whether the game resolution was changed.</returns>
+		bool ResolutionChanged() const { return m_ResolutionChanged; }
+
+		/// <summary>
+		/// Switches the game resolution to the specified dimensions.
+		/// </summary>
+		/// <param name="newResX">New width to resize to.</param>
+		/// <param name="newResY">New height to resize to.</param>
+		/// <param name="upscaled">Whether the new resolution should be upscaled.</param>
+		/// <param name="displaysAlreadyMapped">Whether to skip mapping displays because they were already mapped elsewhere.</param>
+		void ChangeResolution(int newResX, int newResY, bool upscaled, bool displaysAlreadyMapped = false);
+
+		/// <summary>
+		/// Switches to a new resolution multiplier.
+		/// </summary>
+		void ChangeResolutionMultiplier();
+
+		/// <summary>
+		/// Completes the resolution change by resetting the flag.
+		/// </summary>
+		void CompleteResolutionChange() { m_ResolutionChanged = false; }
+#pragma endregion
+
+#pragma region Concrete Methods
+		/// <summary>
+		/// Handles window events coming from the SDL event queue.
+		/// </summary>
+		/// <param name="windowEvent">The SDL event to handle.</param>
+		void HandleWindowEvent(const SDL_Event &windowEvent);
+
+		/// <summary>
+		/// Updates the state of this WindowMan.
+		/// </summary>
+		void Update();
+
+		/// <summary>
+		/// Clears the primary renderer, or all the renderers if in multi-display fullscreen.
+		/// </summary>
+		void ClearRenderer() const;
+
+		/// <summary>
+		/// Copies the BackBuffer32 content to GPU and shows it on screen.
+		/// </summary>
+		void UploadFrame() const;
+#pragma endregion
+
+	private:
+
+		std::shared_ptr<SDL_Window> m_PrimaryWindow; //!< The main window.
+		std::shared_ptr<SDL_Renderer> m_PrimaryRenderer; //!< The main window renderer, draws to the main window.
+		std::unique_ptr<SDL_Texture, SDLTextureDeleter> m_PrimaryTexture; //!< The main window renderer's drawing surface.
```

Comment:
Why is this one a unique_ptr, but the others are shared_ptrs?

Reply:
I have no idea what you're saying lol, please explain more/better.
Is it cause primarywindow is included in the multidisplaywindows vector?

Reply:
Yes.